ajax.php

```diff
@@ -76,11 +76,21 @@
 			$sObjClass = $aData['sContainerClass'];
 			$iObjId = $aData['sContainerId'];
 			$oObj = $aObjByClassAndId[$sClass][$iObjId];
+
 			if ($oObj !== null)
 			{
 				$oDoc = $oObj->Get($aData['sBlobAttCode']);
 				$sFileName = $oDoc->GetFileName();
 				$sIcon = utils::GetAbsoluteUrlAppRoot().AttachmentPlugIn::GetFileIcon($sFileName);
+				$oAttachment = MetaModel::GetObject('Attachment', $iObjId);
+				$sNameUploader = utils::HtmlEntities($oAttachment->Get('contact_id_friendlyname'));
+				$sTimeStampUpload = $oAttachment->Get('creation_date');
+				$sFileFormattedSize = $oDoc->GetFormattedSize();
+				$sFilename = utils::HtmlEntities($sFileName);
+				$sFileName = '<b>'.$sFileName.'</b>';
+				$sFileName .= $sNameUploader != '' ? ' '.Dict::S('UI-emry-attachment-by').' '.$sNameUploader : '';
+				$sFileName .= $sTimeStampUpload != '' ? ' '.Dict::S('UI-emry-attachment-at').' '.$sTimeStampUpload : '';
+				$sFileName .= ' - '.$sFileFormattedSize;
```

Comment:
Mind to use `utils::HtmlEntities()` for the uploader name as well as it could contain special chars.

_Note: Since iTop 3.0 we rather use `utils::EscapeHtml()`, but as this extension must remain compatible with iTop 2.7 we can't use it here._

Also, I would feel more comfortable if we rename `$sFilename` as it is something more now. "$sAttachementLabel" or something like that.

Finally, we can't insert the "at" and "by" like that as in some language it might be a totally different arrangement.
The best approach (to me) would be to have different dict entries for the differents combinaisons. It might be more redundant but also more flexible for all languages.

Reply:
Thank you, will be done. A question about this part:

> Finally, we can't insert the "at" and "by" like that as in some language it might be a totally different arrangement. The best approach (to me) would be to have different dict entries for the differents combinaisons. It might be more redundant but also more flexible for all languages.

If that needs to happen: Do you think, it's ok to move all the code for "making" the label into an own function in ajax.php somewhere above, as I think, there will be quite a few more ifs and elses and the block of code we are talking about already is a bit big?

Reply:
Yeah, it would be nice if there would be a method that was given the attachment and gives the correct text in return..

Reply:
It can be in a method yes, but it must use dict entries no matter what so it can arranged correctly in any language.
Here is what I had in mind:
```
$sFilename = utils::HtmlEntities($sFileName);
$bHasUploaderName = $sNameUploader != '';
$bHasUploadTimestamp = $sTimeStampUpload != '';
if ($bHasUploaderName && $bHasUploadTimestamp) {
	$sDictEntryCode = 'A';
} elseif ($bHasUploaderName && !$bHasUploadTimestamp) {
	$sDictEntryCode = 'B';
} elseif (!$bHasUploaderName && $bHasUploadTimestamp) {
	$sDictEntryCode = 'C';
} else {
	$sDictEntryCode = 'D';
}
$sAttachementLabel = Dict::Format($sDictEntryCode, $sFilename, $sNameUploader, $sTimeStampUpload, $sFileFormattedSize);
```

With dict entries being like
```
'A' => '<b>%1$s</b> by %2$s at %3$s - %4$s',
'B' => '<b>%1$s</b> by %2$s - %4$s',
'C' => '<b>%1$s</b> at %3$s - %4$s',
'D' => '<b>%1$s</b> - %4$s',
```

Reply:
To reduce the number of variants, I think only the filename and date, would be enough?

Reply:
Can you give an example of that you mean? I'm not sure to understand.

Reply:
![](https://user-images.githubusercontent.com/21227916/213553841-020bd28b-9828-4802-a690-f410b7999da2.png)

Reply:
It's up to Martin really, it's fine by me in both cases.
Any thoughts @v-dumas ?